sycl/include/CL/sycl/multi_ptr.hpp

```diff
@@ -44,13 +44,15 @@ template <typename ElementType, access::address_space Space> class multi_ptr {
   multi_ptr() : m_Pointer(nullptr) {}
   multi_ptr(const multi_ptr &rhs) = default;
   multi_ptr(multi_ptr &&) = default;
-  multi_ptr(pointer_t pointer) : m_Pointer(pointer) {}
 #ifdef __SYCL_DEVICE_ONLY__
+  multi_ptr(pointer_t pointer) : m_Pointer(pointer) {}
+#endif
```

Comment:
I think we should apply the same change to `operator=`.

Reply:
Done.